xonsh/jobs.py

```diff
@@ -87,18 +87,39 @@ def _set_pgrp(info):
 
     _shell_pgrp = os.getpgrp()
 
-    _block_when_giving = (signal.SIGTTOU, signal.SIGTTIN, signal.SIGTSTP)
-
-    def _give_terminal_to(pgid):
-        # over-simplified version of:
-        #    give_terminal_to from bash 4.3 source, jobs.c, line 4030
-        # this will give the terminal to the process group pgid
-        if _shell_tty is not None and os.isatty(_shell_tty):
-            if not ON_CYGWIN:
+    _block_when_giving = (signal.SIGTTOU, signal.SIGTTIN,
+                          signal.SIGTSTP, signal.SIGCHLD)
+
+    # _give_terminal_to is a simplified version of:
+    #    give_terminal_to from bash 4.3 source, jobs.c, line 4030
+    # this will give the terminal to the process group pgid
+    if ON_CYGWIN:
+        import ctypes
+        _libc = ctypes.CDLL('cygwin1.dll')
```

Comment:
*Touche!*  I don't know enough about cygwin to know whether this will always be the libc implementation that is used.  If so, it is fine as it is.  I don't consider this blocking :)  Just a suggestion